slides/_fstrings_magic_config.qmd

```diff
@@ -0,0 +1,232 @@
+# Writing better (Python) code
+
+## f-strings {.smaller}
+
+A better way to format strings since Python 3.6\
+Not catching on because of self-teaching from old code.
+
+Strings are prepended with an `f` allowing variables to be used in-place:
+
+```python {code-line-numbers="|4-5|7-8|10-11"}
+name = "electron"
+mass = 9.1093837015E-31
+
+# modulo
+print("The mass of an %s is %.3e kg." % (name, mass))
+
+# format
+print("The mass of an {} is {:.3e} kg.".format(name, mass))
+
+# f-string
+print(f"The mass of an {name} is {mass:.3e} kg.")
+```
+
+f-strings can take expressions:
+
+```python
+print(f"a={a} and b={b}. Their product is {a * b}, sum is {a + b}, and a/b is {a / b}.")
+```
+
+See [Real Python](https://realpython.com/python-f-strings/) for more information.
+Note: pylint W1203 recommends against using f-strings in logging calls.
+
+## Naming Standards
+
+It may seem inconsequential, but carefully naming variables and methods can improve the
+readability of code massively and can help to make code self documenting.
+
+A few naming tips and conventions:
+
+- The name should show the intention, think about how someone else might read it
+- Use pronounceable names e.g. `mass` not `ms`, `stem` not `stm`
+- avoid abbreviations and single letter variable names where possible
+- One word per concept e.g. choose one of `put`, `insert`, `add` in the same code base
+- Use names that can be searched
+- Describe content rather than storage type
+- Plurals to indicate groups, e.g. a list of dog objects would be `dogs`, not `dog_list`
+- Keep it simple and use technical terms where appropriate
+- Use explaining variables
```

Comment:
These are all fantastic, I would encourage a follow-up slide to demonstrate some examples, similar to the "magic numbers" section below.